activerecord/lib/active_record/database_configurations.rb

```diff
@@ -70,6 +74,16 @@ def configs_for(env_name: nil, spec_name: nil, name: nil, include_replicas: fals
       end
     end
 
+    def check_prepared_statements_deprecation(config)
+      adapter = config.adapter.to_sym
+
+      if adapter == :mysql2 && !config.configuration_hash.key?(:prepared_statements)
```

Comment:
Why not check this in the mysql2 adapter where we set the value?

Reply:
Thanks for the review, Rafael. We moved it to the mysql2 adapter 👍